packages/common/src/createContract.ts

```diff
@@ -150,13 +158,43 @@ export function createContract<
               }
             >getFunctionParameters(parameters as any);
 
-            const request = {
+            const functionSignature = getAbiItem<Abi, string>({
+              abi,
+              name: functionName,
+            }) as AbiFunction;
+            const functionSelector = getFunctionSelector(functionSignature);
+            if (!functionSignature) throw new Error(`Unable to get function signature for ${functionName}`);
```

Comment:
```suggestion
            });
            if (!functionSignature) throw new Error(`Unable to get function signature for ${functionName}`);
            const functionSelector = getFunctionSelector(functionSignature);
```
casting `as AbiFunction` and hiding the possible `undefined` seems unwise if we're already checking for the `functionSignature` later, can just move up the conditional

Reply:
furthermore, do you think throwing is the right outcome here? an alternative is to `console.warn` and let this pass through as a regular write request (i.e. don't route it through `callFrom`)

Reply:
`getFunctionSelector` will throw slicing error if `functionSignature` is an unexpected type, so might be better to throw a clearer message earlier. This case is probably unlikely to happen since the abi should contain the signature for the given functionName.

Reply:
would we want to try/catch in that case?

Reply:
that seems equivalent to throwing early here since the only case that `getFunctionSelector ` will throw is the input being either null or undefined. Do we have a preference to use try catch instead?